### ISSUE TITLE
List of port: undefined

### ISSUE THREAD

**Issue by bjwelker** [ioBroker/ioBroker.rflink#8]
Ich habe RFLink auf meinem Docker ioBroker installiert. Das Gerät wird unter /dev/ richtig angezeigt. Der Container läuft priviligiert somit sollten die Geräte auch auf dem Container da sein. Leider kommt im Log List of port: undefined und in der Plugin Config ist im Dropdown auch nur ganz oft undefined

Irgendwer ne Idee?

**Comment by Apollon77** [comment]
Läuft iobroker unter Root im Container?

**Comment by bjwelker** [comment]
Yep läuft privilegiert

**Comment by Apollon77** [comment]
Es gibt eine neue rflink Version im github die ich heute zum testen ins Forum stelle. Versuch die mal.

**Comment by Apollon77** [comment]
Feedback?

**Comment by chucky2002** [comment]
Habe das gleiche Problem wie bjwelker. Der Fehler tritt auch mit der aktuellen Version von Github auf.
Gern liefere ich weitere Details zu dem Problem.

**Comment by chucky2002** [comment]
Läuft in einem Docker-Container auf der Synology NAS.  Werden füp das Device (bei mir  /dev/ttyACM0) spezielle Berechtigungen benötigt? Müssen spezielle Voraussetzungen geschaffen werden damit das Device aufgelistet wird.
Hatte zuvor eine iobroker Installation auf einem Raspberry und dort hat das Modul rflink super funktioniert (war glaub ich am Anfang dieses Jahres).

**Comment by Apollon77** [comment]
Die Frage müsste wohl eher an das serialport Projekt gehen warum die v6.x in einem Docker container keine devices listen kann

**Comment by chucky2002** [comment]
Gibt es ein Kommando mit dem ich die Aufzählung auf der Kommandozeilenebene testen kann? Bin nicht so gut in JavaScript und npm. Dann habe ich auch ein paar Details die ich zu einem möglichen Fehler einreichen kann.

**Comment by Apollon77** [comment]
Das hier wäre zwar die v7 von serialport (Adapter nutzen v6), aber vllt kann man damit mal starten:

https://serialport.io/docs/en/guide-cli

**Comment by nkleber78** [comment]
Hallo zusammen,
habe / hatte das selbe problem. Es fehlt im IOBroker container "udev" habe bei mir mit "apt-get install udev" installiert. Dann werden sofort die Devices gefunden und richtig dargestellt. Es wäre sicherlich sinnvoll dass bei der Installation des Adapters dies zukünftig geprüft wird und ggf. nachinstalliert. 
Weiters ist noch wichtig dass es derzeit nicht möglich ist direkt das device z.b. /dev/ttyUSB0 in den docker container unter /dev/ttyUSB0 zu mappen. Daher habe ich, da ich nicht privilegiert laufen will, das device auf /opt/iobroker/dev/ttyUSB0 gemappt und dann einen softlink ln -s /opt/iobroker/dev/ttyUSB0 /dev/ttyUSB0 gemacht. Dann läuft das System. Auch hier die Frage ob der Adapter für diese Situation verbessert werden könnte

**Comment by Apollon77** [comment]
> Es wäre sicherlich sinnvoll dass bei der Installation des Adapters dies zukünftig geprüft wird und ggf. nachinstalliert.
Nein. Der Adapter läuft generisch für mehrere Betriebssysteme und es geht auch alles in normalen Linux-Umfeldern. Wenn jemand Docker nutzt muss er so etwas wissen. Ich bin dafür es ggf zu dokumentieren

**Comment by nkleber78** [comment]
> Nein. Der Adapter läuft generisch für mehrere Betriebssysteme und es geht auch alles in normalen Linux-Umfeldern. Wenn jemand Docker nutzt muss er so etwas wissen. Ich bin dafür es ggf zu dokumentieren

Ok, in der Doku würde sicherlich schon mal helfen. Wenn man eine Prüfung auf vorhanden sein von udev machen würde, und den User informiert wenn es nicht vorhanden ist, wäre das natürlich genial.
Dass ein normaler Docker user das weiß / wissen kann, kann ich mir nicht vorstellen, dazu ist das viel zu komplex für die meisten User. Und es sind ja nicht nur profis die Docker verwenden, sondern jeder Heim user der Synology hat... Wie gesagt mindestens Doku, prüfen und informieren wäre genial

Danke
